lib/process-base.js

```diff
@@ -0,0 +1,29 @@
+module.exports = class ProcessBase {
+    constructor(pid) {
+        if (this.constructor === ProcessBase) {
+            throw new Error('Cannot instantiate abstract class');
+        }
+
+        this.pid = pid;
+    }
+
+    attach() {
+        throw new Error('Abstract method cannot be called');
+    }
+
+    detach() {
+        throw new Error('Abstract method cannot be called');
+    }
+
+    /* generator */ iterateMemoryRegions() {
```

Comment:
Why comment instead of iterator?

Reply:
Because if it's really a generator, then:
```class MyProcess extends BaseProcess { }
const proc = new MyProcess();
const iterator = proc.iterateMemoryRegions();```
This won't throw until `iterator.next()` is called.
The comment is to indicate that child classes should make this a generator.